include/seqan3/core/detail/strong_type.hpp

```diff
@@ -0,0 +1,484 @@
+        return derived_t{get() ^ other.get()};
+    }
+
+    //!\brief Adds bitwise not operator to the strong type.
+    constexpr auto operator~()
+        //!\cond
+        requires ((skills & strong_type_skill::bitwise_not) != strong_type_skill::none)
+        //!\endcond
+    {
+        return derived_t{~get()};
+    }
+    //!\}
+
+    /*!\name Bitwise shift operators.
+     * \brief Only available if the corresponding skills from seqan3::detail::strong_type_skill are added and
+     *        the underlying type supports this operation.
+     * \{
+     */
+
+    //!\brief Adds bitwise left shift operator to the strong type.
+    constexpr auto operator<<(strong_type const & other)
+        //!\cond
+        requires ((skills & strong_type_skill::bitwise_lshift) != strong_type_skill::none)
+        //!\endcond
+    {
+        return derived_t{get() << other.get()};
+    }
+
+    //!\brief Adds bitwise left shift operator to the strong type.
+    constexpr auto operator<<(integral_concept const shift)
+        //!\cond
+        requires ((skills & strong_type_skill::bitwise_lshift) != strong_type_skill::none)
+        //!\endcond
+    {
+        return derived_t{get() << shift};
+    }
+
+    //!\brief Adds bitwise right shift operator to the strong type.
+    constexpr auto operator>>(strong_type const & other)
+        //!\cond
+        requires ((skills & strong_type_skill::bitwise_rshift) != strong_type_skill::none)
+        //!\endcond
+    {
+        return derived_t{get() >> other.get()};
+    }
+
+    //!\brief Adds bitwise right shift operator to the strong type.
+    constexpr auto operator>>(integral_concept const shift)
+        //!\cond
+        requires ((skills & strong_type_skill::bitwise_rshift) != strong_type_skill::none)
+        //!\endcond
+    {
+        return derived_t{get() >> shift};
+    }
+    //!\}
+
+    /*!\name Logical operators.
+     * \brief Only available if the corresponding skills from seqan3::detail::strong_type_skill are added and
+     *        the underlying type supports this operation.
+     * \{
+     */
+
+    //!\brief Adds logical and operator to the strong type.
+    constexpr auto operator&&(strong_type const & other)
+        //!\cond
+        requires ((skills & strong_type_skill::logical_and) != strong_type_skill::none)
+        //!\endcond
+    {
+        return derived_t{get() && other.get()};
+    }
+
+    //!\brief Adds logical or operator to the strong type.
+    constexpr auto operator||(strong_type const & other)
+        //!\cond
+        requires ((skills & strong_type_skill::logical_or) != strong_type_skill::none)
+        //!\endcond
+    {
+        return derived_t{get() || other.get()};
+    }
+
+    //!\brief Adds logical not operator to the strong type.
+    constexpr auto operator!()
+        //!\cond
+        requires ((skills & strong_type_skill::logical_not) != strong_type_skill::none)
+        //!\endcond
+    {
+        return derived_t{!get()};
+    }
+    //!\}
```

Comment:
on the above operators, why do you not have the return type be `derived_t` instead of doing `auto` in the signature and then creating the type inside?